app/models/miq_worker.rb

```diff
@@ -253,6 +253,18 @@ def self.worker_settings(options = {})
     fetch_worker_settings_from_server(MiqServer.my_server, options)
   end
 
+  def self.reload_worker_settings
+    # By default worker settings are reloaded periodically by the worker runner
+    return if rails_worker?
+
+    settings = {}
+    worker_settings_paths.to_a.each do |settings_path|
+      settings.store_path(settings_path, Settings.to_hash.dig(*settings_path))
+    end
+
+    restart_workers unless Vmdb::Settings::HashDiffer.changes(worker_settings, settings).empty?
```

Comment:
So `worker_settings` here is the cached old values?  This doesn't include the additional settings in `worker_settings_paths` though right?

Reply:
The `worker_settings` are retrieved from the server https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_worker.rb#L252-L254 (I think they are cached?). And yes this does not include the additional settings from `worker_settings_paths`

Reply:
Okay `worker_settings` is something specific, and `worker_settings_paths` are "in-addition-to" the base `worker_settings`.

What I was picturing was when we have the new full set of settings and the cached old full set of settings (we need to find the best place for this), get the list of paths that were changed and compare to the `worker_settings_paths` and `worker_settings` paths.